src/Statement.php

```diff
@@ -95,7 +127,17 @@ public function bindValue($name, $value, $type = null)
      */
     public function bindParam($name, &$var, $type = PDO::PARAM_STR, $length = null)
     {
-        return $this->stmt->bindParam($name, $var, $type, $length);
+        $args = func_get_args();
+        // func_get_args() returns copy, not reference
+        $args[1] = &$var;
+
+        if (call_user_func_array([$this->stmt, __FUNCTION__], $args)) {
```

Comment:
same as previous comment,
may you use a direct call to `$this->stmt->bindParam()` ?
bonus point: you'll get `&$var` as a real reference!

Reply:
This way it allows to use the fourth parameter `$driverOptions` in case a `PDOStatement` is wrapped, see https://github.com/doctrine/dbal/blob/9efbf7f3e260442ceefe4aec163c18b193c8e0c2/lib/Doctrine/DBAL/Driver/PDOStatement.php#L110. Though probably nobody needs it ;-) So it is up to you to decide which approach to use.

Reply:
here too, as above!

Reply:
Had a look at it again: Because we are actually wrapping a `\Doctrine\DBAL\Statement` and never a `\Doctrine\DBAL\Driver\PDOStatement` the fourth parameter makes no sense here.